### PR TITLE
Change transferleanring metrics datastructure

### PR DESCRIPTION
**Pull request by mo-sameh**
The data structure used for storing the metrics has been changed to a long format. This has made the metrics module, and particularly the metric manager, much simpler and more general. Additionally, I have adapted train.py and the plotting functions in the fine-tuning notebook to use the new data structure.

### CODE REVIEW

File: alphadia/transferlearning/metrics.py [hunk 1]
```diff
@@ -5,67 +5,55 @@
...
+        pd.DataFrame
+            A pandas dataframe containing the stats in a long format.
 
         """
-        new_stats.index = [epoch]
-        new_stats.index.name = "epoch"
-        self.stats = (
-            new_stats if self.stats is None else pd.concat([self.stats, new_stats])
-        )
+        long = []
```
Comment (GeorgWa) [inline alphadia/transferlearning/metrics.py:39]: maybe this could be done more pythonic with `pandas.melt()`

File: alphadia/transferlearning/metrics.py [hunk 2]
```diff
@@ -5,67 +5,55 @@
...
     """
 
     def __init__(self, columns: List[str]):
         self.columns = columns  # a list of column names for the stats dataframe
-        self.stats = (
-            None  # Stats is a pandas dataframe that stores the test metric over time
-        )
 
-    def _update_stats(self, new_stats: pd.DataFrame, epoch: int):
+    def _to_long_format(
```
Comment (GeorgWa) [inline alphadia/transferlearning/metrics.py:16]: Would it also be possible to calculate the metrics immediately as long format without the need to call `_to_long_format()` ? I guess this is due to peptdeep metrics being calculated in dense format?

File: alphadia/transferlearning/metrics.py [hunk 3]
```diff
@@ -79,22 +67,29 @@ def calculate_test_metric(self, test_input: dict, epoch: int):
...
+        property_name : str
+            The name of the property. e.g. "charge", "rt"
+
         Returns
         -------
-        pd.DataFrame
-            A pandas dataframe containing the test metric at the given epoch.
+        list[dict]
+            A list of dictionaries containing the test metric at the given epoch in the long format.
+            e.g. [{"dataset": "train", "epoch": 1, "property": "charge", "metric_name": "accuracy", "value": 0.9}]
```
Comment (GeorgWa) [inline alphadia/transferlearning/metrics.py:80]: I think we can already have them as pandas dataframes right from the begininng. No need to have the list of dicts pattern.

File: alphadia/transferlearning/metrics.py [hunk 4]
```diff
@@ -5,67 +5,55 @@
...
+                        "property": property_name,
+                        "metric_name": column,
+                        "value": row[1][column],
+                    }
+                    long.append(subject)
+        return long
 
-    def calculate_test_metric(self, test_input: dict, epoch: int):
+    def calculate_test_metric(
+        self, test_input: dict, epoch: int, dataset: str, property_name: str
```
Comment (GeorgWa) [inline alphadia/transferlearning/metrics.py:55]: Quick question :D We pass a dict because the MS2 metrics need acces to the `psm_df`?

File: alphadia/transferlearning/train.py [hunk 5]
```diff
@@ -315,6 +315,8 @@ def _test_ms2(
             The matched fragment intensity dataframe.
         metric_accumulator : MetricManager
             The metric manager object.
+        dataset : str
```
Comment (GeorgWa) [inline alphadia/transferlearning/train.py:318]: maybe dataset is not precise enough. What do you think about `data_split` or `data_subset` instead? Ofc for the whole PR

File: alphadia/transferlearning/train.py [hunk 6]
```diff
@@ -525,26 +541,47 @@ def _test_rt(
...
+                    property_name="rt",
+                )
+                loss = val_metrics[0]["value"]
+                continue_training = self.early_stopping.step(loss)
+                logger.progress(
+                    f" Epoch {epoch:<3} Lr: {current_lr:.5f}   Training loss: {epoch_loss:.4f}   validation loss: {val_metrics[0]['value']:.4f}"
+                )
+            else:
+                logger.progress(
+                    f" Rt model tested on {dataset} dataset with the following metrics:"
```
Comment (GeorgWa) [inline alphadia/transferlearning/train.py:576]: (nit) RT

File: nbs/tutorial_nbs/finetuning.ipynb
Comment (GeorgWa) [inline nbs/tutorial_nbs/finetuning.ipynb:1]: SHould the whole notebook be removed?

**Review by GeorgWa — APPROVED** [review]
Really nice, I like the long format and it's use across test, train, validation very much!
LGTM with some minor changes.

File: nbs/tutorial_nbs/finetuning.ipynb
Comment (GeorgWa) [reply nbs/tutorial_nbs/finetuning.ipynb:1]: Nevermind, I missread the diff :D
